lib/dataspec/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+use serde::{Deserialize, Serialize};
+
+use histogram::CompactHistogram;
+
+#[derive(Serialize, Deserialize)]
```

Comment:
All of these structs should at least be deriving `Clone` and `Debug`. `Copy` is optional and depends on whether you think the struct will need to have non-copy fields in the future (e.g. strings).

```suggestion
#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
```

Reply:
`CompactHistogram` doesn't implement any of those, so I'm going to hold off on adding these here until those are added on the next version bump.